### PR TITLE
Fix JsonInclude NON_EMPTY on collection

### PR DESCRIPTION
**Pull request by cgendreau**
* Added JsonInclude Test
* Fixed JsonInclude.Non_Empty
* isDefaultNodeValue method fix

### CODE REVIEW

**Comment by cgendreau** [comment]
Fix Issue #790

File: crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java [hunk 1]
```diff
@@ -186,7 +186,7 @@ private boolean isNodeIncluded(ResourceField field, JsonNode node) {
 
 	protected boolean isDefaultNodeValue(JsonNode node) {
 		return isNullNodeValue(node) || (node.isObject() || node.isArray()) && node.size() == 0
-				|| node.asText().isEmpty() || node.isNumber() && node.asDouble() == 0d;
+				|| node.isValueNode() && node.asText().isEmpty() || node.isNumber() && node.asDouble() == 0d;
 	}
```
Comment (remmeier) [inline crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java:190]: is there possibly a more efficient way? I would expect that asText to be rather inefficient for non primitve fields.

Comment (Jkeyuk) [reply crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java:190]: I believe the intention of the author was to skip the ` node.asText().isEmpty()` for fields that were not primitive. As I believe `node.isValueNode()` will return false for JSON types that are arrays or objects.

Comment (remmeier) [reply crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java:190]: ok thx!

**Pull request merged**